### ISSUE TITLE
[Bridge RA-HPG v1.0 Run] Execute preregistered relation-capacity seed42 validation experiment

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#57]
## Objective

Execute exactly one **RA-HPG v1.0 relation-capacity seed-42 FER2013 train/validation run** after independent implementation review and merge of PR #56.

Primary scientific question:

> Does the preregistered RA-HPG relation-capacity block — dynamic per-block graph recomputation + relation-aware scalar-gated aggregation + 4x FFN — recover materially more fit and validation performance than accepted CF-HPG v1.2 while keeping generalization gaps controlled?

This issue authorizes **one scientific training/validation attempt only**. It does not authorize test access, hyperparameter search, resume, automatic retry, architecture changes, ablation decomposition, or any follow-up RA-HPG version.

Important interpretation boundary: this is one **compound relation-capacity block intervention**. A positive result may support the block as a whole; it must not be attributed causally to dynamic topology, relation gating, or FFN expansion individually without a later ablation.

## Exact authorized source

Run only from exact merged `main` commit:

`f8a183520ab1ef7d949e66b48ee7ce7e352f1cb6`

Reviewed PR #56 head:

`47cae34f85715beb13540714d18df35ce3278ebf`

Implementation parent:

`8468dfe40ae2e6229c835b3356bb61eac1f22c77`

Registered candidate:

`research/candidates/tf_ra_hpg_v1_relation_aware/`

Registered model source SHA-256:

`a312e7bc9fc00df45dc11e897ba99a09eb24484ecd7a9c03748a3bb4e7ba6500`

Registered data source SHA-256:

`e8b586aafc6fbad74d913cb08d25a61432c0e55173a60621597f4c5a8d091776`

Registered graph source SHA-256:

`fd425766f8db4d53f87edca0bebb88ff476c43938fff6a507c0687a0efb2d8ec`

Registered lifecycle source SHA-256:

`ef24467f132df342333435c48251fc115b99bec622329bb725b7d58ab55f787b`

Expected exact model identity:

- parameters `626987`
- trainable variables `74`
- Keras variables `84`

If any source hash or identity differs, STOP BEFORE TRAINING.

### Representation

- input `48x48x1` grayscale
- exact scale `x/127.5 - 1`
- non-overlap `4x4` patches
- `12x12 = 144` fine nodes
- raw width `16`
- `Dense(16->96) -> GELU -> Dense(96->96)`
- generic normalized 2D grid centers -> `Dense(2->96)`
- additive position fusion
- `LayerNorm -> GELU -> Dropout(0.10)`

### Frozen topology function

For current block input nodes:

`A = spatial_8_neighbor_grid UNION cosine_top4(current_nodes)`

- directed
- no self edges
- exact top-k `4`
- union/dedup
- no anatomy/semantic edges

Recompute this same topology immediately before **every** graph block from that block's current unnormalized input:

- fine block 1 -> fresh `A_f1`
- fine block 2 -> fresh `A_f2`
- coarse block 1 -> fresh `A_c1`
- coarse block 2 -> fresh `A_c2`

### Relation-aware block

Exactly 2 fine blocks at `D=96` and 2 coarse blocks at `D=128`.

For input `x` and fresh adjacency `A`:

1. `z_i = LayerNorm(x_i)`
2. `r_ij = z_j - z_i`
3. `a_ij = sigmoid(Dense_D_to_1(r_ij))`, scalar biased gate
4. `degree_i = sum_j 1[A_ij]`
5. `m_i = sum_j 1[A_ij] * a_ij * r_ij / max(degree_i,1)`
6. graph mixer: `concat[z_i||m_i] -> Dense(2D->D) -> GELU -> Dense(D->D) -> Dropout(0.15)` + residual
7. FFN: `LayerNorm(D) -> Dense(D->4D) -> GELU -> Dense(4D->D) -> Dropout(0.15)` + residual

No softmax attention, no vector gate, no max aggregation, no BatchNorm, no extra blocks.

### Hierarchy/readout

After two fine blocks:

- exact parameter-free `2x2` arithmetic mean
- `144 -> 36` nodes
- `Dense(96->128) -> LayerNorm -> GELU`
- exactly two coarse blocks
- coarse-only global mean + max -> `[B,256]`
- `LayerNorm(256) -> Dense(256->128) -> GELU -> Dropout(0.20) -> Dense(128->7)`

No v1.3 multiscale readout.

## Dataset boundary

Dataset signature:

`fer2013_train28709_val3589_test3589`

Allowed only:

- train exactly `28709`
- validation exactly `3589`

Absolutely forbidden:

- open/read/hash/load `test.csv`
- test labels/predictions/metrics/confusion matrix
- test smoke/debug/model selection
- copying/renaming test paths to bypass guards

Any lexical/path guard failure => STOP.

## Frozen training configuration

Exactly accepted CF-HPG v1.2 semantics:

- seed `42`
- AdamW
- LR `3e-4`
- weight decay `5e-4`
- global clipnorm `1.0`
- batch `64`
- max epochs `100`
- linear warmup `5`
- cosine decay to `1e-6`
- CE from logits
- label smoothing `0.05`
- checkpoint = earliest strict max `val_accuracy`
- early stopping monitor = `val_loss`
- patience `15`
- min_delta `0`

Train-only augmentation remains frozen from v1.2. Validation and clean-train evaluation have augmentation OFF.

No MixUp, CutMix, SAM, TTA, class weights, oversampling, focal loss, alternate optimizer/LR/WD/batch/seed, DropEdge, or DropMessage.

## Runtime preflight

Use the same Kaggle runtime class as accepted v1.1-v1.3 when available:

- Python `3.12.13`
- TensorFlow `2.20.0`
- Keras `3.13.2`
- 1 x `Tesla P100-PCIE-16GB`
- CUDA `12.5.1`
- cuDNN `9`
- mixed precision policy `float32`
- external XLA/JIT empty/off

Before launch record exact values. If the selected Kaggle runtime does not match this registered runtime class, STOP before the training command and report preflight mismatch; the scientific attempt is not consumed.

TensorFlow-internal XLA cluster compilation is allowed only as a recorded runtime note; do not externally enable XLA/JIT or mixed precision.

## Exact authorized command

From exact detached checkout `f8a183520ab1ef7d949e66b48ee7ce7e352f1cb6`:

```bash
python -m research.candidates.tf_ra_hpg_v1_relation_aware.train_validation_only \
  --train-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --val-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv \
  --output-root /kaggle/working/tf_ra_hpg_v1_relation_aware_seed42
```

Do not add/change flags.

## One-attempt rule

Checkout/auth/source/runtime/dataset preflight stops **before** the training command starts do not consume the attempt.

Once the exact training command starts, the one authorized scientific attempt is consumed. After completion, early stopping, crash, OOM, CUDA failure, Kaggle censor, or other runtime failure: STOP. Do not rerun, resume, change batch size, or modify source/config.

Any second attempt requires a new written amendment justified by technical invalidity, never performance.

## Locked comparison reference: accepted CF-HPG v1.2

- clean-train accuracy `0.614894284022432`
- clean-train macro-F1 `0.5614369765915708`
- validation accuracy `0.5806631373641683`
- validation macro-F1 `0.5238975323290902`

Exact result field:

`deltas_vs_cf_hpg_v1_2_pp`

Exact deltas:

- `delta_val_accuracy_pp = 100*(RA_val_acc - 0.5806631373641683)`
- `delta_val_macro_pp = 100*(RA_val_macro - 0.5238975323290902)`
- `delta_clean_train_accuracy_pp = 100*(RA_train_acc - 0.614894284022432)`
- `delta_clean_train_macro_pp = 100*(RA_train_macro - 0.5614369765915708)`

### Context-only LAP-GNN anchors

These are **not** selection criteria and do not alter the registered decision label:

- LAP seed42 best val accuracy `0.6319308999721371`
- LAP val macro-F1 at that selected val-accuracy checkpoint `0.5938407974340496`
- LAP seed42 best val macro-F1 `0.601166548701511` at a different epoch
- LAP train/val macro gap at best-macro epoch `15.511397995653287 pp`

Report contextual difference from the first two same-checkpoint anchors after the RA result, but never use LAP to choose a checkpoint or alter the preregistered decision.

## Frozen decision precedence

1. `RA_HPG_V1_STRETCH_PASS`
2. `RA_HPG_V1_PASS`
3. `RELATION_OVERFIT_SHIFT`
4. `RELATION_STRONG_SIGNAL`
5. `RELATION_PARTIAL_SIGNAL`
6. `RELATION_FIT_WITHOUT_VAL_GAIN`
7. `RELATION_UNDERFIT_REMAINS`
8. `RELATION_INCONCLUSIVE`

### RA_HPG_V1_STRETCH_PASS

All:

- val accuracy >= `0.7000`
- val macro-F1 >= `0.6700`
- accuracy gap <= `8 pp`
- macro gap <= `8 pp`

### RA_HPG_V1_PASS

All:

- val accuracy >= `0.6500`
- val macro-F1 >= `0.6200`
- accuracy gap <= `8 pp`
- macro gap <= `8 pp`

### RELATION_OVERFIT_SHIFT

- clean-train accuracy improvement vs v1.2 >= `5 pp`
- AND either clean accuracy gap > `10 pp` or clean macro gap > `10 pp`

### RELATION_STRONG_SIGNAL

- validation accuracy improvement vs v1.2 >= `5 pp`
- AND clean-train accuracy improvement vs v1.2 >= `5 pp`
- unless higher-priority label applies

### RELATION_PARTIAL_SIGNAL

- validation accuracy improvement vs v1.2 >= `3 pp` and < `5 pp`
- unless higher priority applies

### RELATION_FIT_WITHOUT_VAL_GAIN

- clean-train accuracy improvement vs v1.2 >= `5 pp`
- validation accuracy improvement vs v1.2 < `3 pp`
- both clean gaps <= `10 pp`
- unless higher priority applies

### RELATION_UNDERFIT_REMAINS

- clean-train accuracy < `0.7000`
- validation accuracy < `0.6200`
- unless higher priority applies

Else `RELATION_INCONCLUSIVE`.

These are single-seed practical diagnostic labels only, not statistical superiority/equivalence claims.

## Predeclared post-result action

After independent evidence review:

- `PASS` or `STRETCH_PASS`: freeze architecture immediately; proceed to confirmatory seeds + minimal graph ablations. No tuning first.
- `RELATION_STRONG_SIGNAL`: if val remains below PASS, allow at most one separately preregistered final refinement only if the fit/generalization pattern identifies a single clear bottleneck.
- `RELATION_PARTIAL_SIGNAL`: deadline decision; no automatic refinement. Research lead decides whether the gain is sufficient for paper use.
- `RELATION_FIT_WITHOUT_VAL_GAIN` or `RELATION_OVERFIT_SHIFT`: capacity was recovered without acceptable generalization; only graph-native regularization may be considered in a new preregistered experiment, never architecture expansion.
- `RELATION_UNDERFIT_REMAINS`: kill the RA-HPG bridge family for the Sep-14 cycle. No further no-CNN patch-graph redesign before submission.
- `RELATION_INCONCLUSIVE`: no automatic follow-up.

## Required evidence

Preserve fresh output root and complete available log. Require at minimum:

- `resolved_config.json`
- `model_summary.txt`
- `checkpoints/best_val_accuracy.keras`
- `checkpoints/best_val_accuracy.metadata.json`
- `validation_only_result.json`
- exact pre-run manifest with source/runtime/model identity
- exact command and process return code
- explicit `test_access=false`

After termination package output root exactly once as:

`/kaggle/working/tf_ra_hpg_v1_relation_aware_seed42_evidence.zip`

Compute and report:

- final ZIP SHA-256
- ZIP bytes
- member count
- outer ZIP integrity
- embedded `.keras` SHA-256
- embedded `.keras` integrity/member count

Do not repack after the final ZIP hash.

## Required final report

Return:

1. exact checkout SHA
2. clean worktree before/after
3. model/data/graph/lifecycle SHA-256
4. params / trainable vars / Keras vars
5. runtime identity
6. exact command
7. process return code
8. epochs completed
9. selected checkpoint epoch
10. clean train samples/accuracy/macro-F1/loss
11. validation samples/accuracy/macro-F1/loss
12. accuracy gap pp
13. macro gap pp
14. exact four deltas vs CF-HPG v1.2
15. context-only differences vs LAP same-checkpoint anchors
16. registered decision label
17. checkpoint metadata
18. early-stop/completion state
19. `test_access=false`
20. evidence ZIP SHA/bytes/member count/integrity
21. embedded `.keras` SHA/integrity/member count
22. warnings/errors/runtime notes

Final valid status:

`RA_HPG_V1_RELATION_AWARE_SEED42_COMPLETE_PENDING_INDEPENDENT_REVIEW`

If technically invalid:

`RA_HPG_V1_RELATION_AWARE_SEED42_INVALID_PENDING_INDEPENDENT_REVIEW`

No test access is authorized by this issue.

**Comment by Irthn1311** [comment]
## Research-lead review status — protocol-level consistency PASS, archive-level review pending

I independently checked the registered Issue #57 contract, exact merged source lineage, and the reported metric arithmetic/decision logic. The report is internally consistent with the preregistration:

- exact authorized source `f8a183520ab1ef7d949e66b48ee7ce7e352f1cb6`;
- registered source hashes and model identity match the Issue #57 contract;
- reported gaps and all four deltas versus accepted CF-HPG v1.2 recompute exactly;
- the reported values satisfy `RELATION_UNDERFIT_REMAINS` under the frozen precedence;
- the reported result is below CF-HPG v1.2 on clean-train accuracy/macro-F1 and validation accuracy/macro-F1.

However, the downloaded Issue #57 ZIP/log/pre-run/wrapper artifacts are not currently attached to the research-lead review channel or committed as accessible evidence, and this GitHub issue currently contains no run-evidence attachment/comment. Therefore I am **not yet marking archive integrity / checkpoint provenance / epoch-history / no-test artifact review as independently accepted**.

Do not rerun, resume, tune, or start another RA-HPG version. The single scientific attempt is consumed. The preregistered post-result action remains in force: if archive-level review confirms the reported result, `RELATION_UNDERFIT_REMAINS` kills the RA-HPG/no-CNN patch-graph redesign branch for the Sep-14 cycle.

Status: `ISSUE57_PROTOCOL_LEVEL_REVIEW_PASS_ARCHIVE_REVIEW_PENDING`.

**Comment by Irthn1311** [comment]
## Independent archive review — ACCEPT

I independently inspected the uploaded Issue #57 evidence archive and companion pre-run/log/wrapper artifacts.

Archive/provenance checks:

- outer ZIP SHA-256: `f911dc3f26ebefa4011c7b7c22a397295bbe0b7e65e1794a5b6f2452fa0ca3e6`
- outer ZIP bytes: `7,250,338`
- outer ZIP members: `9`
- outer ZIP CRC/integrity: PASS
- no CSV members and no member name containing a test split
- uploaded pre-run JSON is byte-identical to the archive copy
- uploaded training log is byte-identical to the archive copy
- embedded checkpoint SHA-256: `be3edfa9df4185c02a408ee688592aeb010fe91ab44f915f60177bc57888d9a3`
- embedded checkpoint bytes: `7,745,835`
- embedded Keras archive integrity: PASS, exactly 3 members (`metadata.json`, `config.json`, `model.weights.h5`)

Pre-run/source/runtime checks match the preregistration exactly:

- HEAD `f8a183520ab1ef7d949e66b48ee7ce7e352f1cb6`
- model/data/graph/lifecycle SHA-256 values match Issue #57
- model identity `626987 / 74 / 84`
- Python `3.12.13`, TensorFlow `2.20.0`, Keras `3.13.2`, 1x Tesla P100, CUDA `12.5.1`, cuDNN `9`, float32, external XLA/JIT empty
- train `28709`, validation `3589`, `test_access=false`
- tracked worktree clean

Training-history checks:

- exactly 100 epoch summaries are present
- max training epoch reached: 100/100; early stopping did not trigger
- rounded epoch-history maximum validation accuracy occurs at human epoch 88; checkpoint metadata selects zero-based epoch 87 under `earliest_strict_max_val_accuracy`
- rounded best validation loss occurs later at epoch 95, consistent with the run continuing to max epoch
- no training exception/traceback invalidating the run
- TensorFlow internal XLA compilation is present only as the already-allowed runtime note

Scientific result independently recomputes to:

- clean-train accuracy `0.6063952070779198`
- clean-train macro-F1 `0.5470866113149493`
- validation accuracy `0.5739760378935637`
- validation macro-F1 `0.508546599576986`
- accuracy gap `3.24191691843561 pp`
- macro gap `3.85400117379633 pp`
- delta train accuracy vs CF-HPG v1.2 `-0.849907694451213 pp`
- delta train macro vs CF-HPG v1.2 `-1.435036527662148 pp`
- delta val accuracy vs CF-HPG v1.2 `-0.668709947060464 pp`
- delta val macro vs CF-HPG v1.2 `-1.5350932752104196 pp`

Under the frozen precedence, the only valid registered label is:

`RELATION_UNDERFIT_REMAINS`

Scientific interpretation boundary:

- the compound relation-capacity intervention did not recover fit or validation performance relative to accepted CF-HPG v1.2 at seed 42;
- because dynamic topology + scalar relation gating + FFN expansion were changed together, this run does not isolate which subcomponent caused the regression;
- the preregistered deadline kill rule therefore applies: no further RA-HPG/no-CNN patch-graph redesign for the Sep-14 submission cycle.

Final status:

`RA_HPG_V1_RELATION_AWARE_SEED42_INDEPENDENT_REVIEW_ACCEPTED`
